client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -117,6 +119,16 @@ public void updateRetentionStreamCut(Map<Stream, StreamCut> streamCuts) {
                synchronizer.getState().getConfig().getRetentionType().toString());
     }
 
+    public UUID getGroupId() {
+        synchronizer.fetchUpdates();
+        return synchronizer.getState().getConfig().getReaderGroupId();
+    }
+
+    public long getGeneration() {
+        synchronizer.fetchUpdates();
```

Comment:
Same here.

Reply:
Removed fetchUpdates from here.

Reply:
I don't see a change. Was it pushed?

Reply:
When calling `deleteReaderGroup` on the controller end we need to pass the generation and readergroupID for which we need their latest versions, hence the `fetchUpdates`. I had initially thought to get the ReaderGroup and then use Impl methods to get these values as I have done above. However, since these getters for the generation and groupID are only being called at one place (in `deleteReaderGroup`) instead of getting the `ReaderGroupImpl` I make the `StateSynchronizer`, call `fetchUpdates` once and then access the latest versions of generation and groupID.